File: src/main/java/com/github/mitsukigoto/lavawalker/init/EnchantmentInit.java
package com.github.mitsukigoto.lavawalker.init;

import com.github.mitsukigoto.lavawalker.LavaWalker;
import com.github.mitsukigoto.lavawalker.enchantment.LavaWalkerEnchantment;
import net.minecraft.enchantment.Enchantment;
import net.minecraft.inventory.EquipmentSlotType;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class EnchantmentInit {
    public static final DeferredRegister<Enchantment> ENCHANTMENTS = DeferredRegister.create(ForgeRegistries.ENCHANTMENTS, LavaWalker.MODID);
    public static final RegistryObject<Enchantment> LAVA_WALKER = ENCHANTMENTS.register("lava_walker", () -> new LavaWalkerEnchantment(Enchantment.Rarity.RARE, EquipmentSlotType.FEET));
}

File: src/main/kotlin/com/github/mikn/lavawalker/init/EnchantmentInit.kt
package com.github.mikn.lavawalker.init;

import com.github.mikn.lavawalker.enchantment.LavaWalkerEnchantment;
import net.minecraft.world.entity.EquipmentSlot;
import net.minecraft.world.item.enchantment.Enchantment;

class EnchantmentInit {
    companion object {
        @JvmField
        val LAVA_WALKER = LavaWalkerEnchantment(Enchantment.Rarity.RARE, EquipmentSlot.FEET)
    }
}
